Improve OTEL docs according to customer feedback (#6195)
getsentry/sentry-docs: src/platforms/common/performance/instrumentation/opentelemetry.mdx

<Include name="beta-note-opentelemetry.mdx" />

<PlatformSection notSupported={["java"]}>

You can configure your [OpenTelemetry SDK](https://opentelemetry.io/) to send traces and spans to Sentry.

</PlatformSection>

<PlatformSection supported={["java"]}>

There are multiple ways of configuring your [OpenTelemetry SDK](https://opentelemetry.io/) to send traces and spans to Sentry.

</PlatformSection>

> ## Preface

Author: Please helps improve this section to make it accurate and easy to understand.

Reviewer: <img width="810" alt="image" src="https://user-images.githubusercontent.com/363802/216025025-4ee328c3-535b-4196-abd3-212680efc091.png">

```
An OpenTelemetry Span can become a Sentry Transaction or Span. The first Span sent through the Sentry SpanProcessor is a Transaction, and any child Span of it is attached to the Transaction by checking the parent Span context. This happens for the OpenTelemetry root Span and any top-most Span in the system. For example, a request from frontend to backend creates an OpenTelemetry root Span and its corresponding Sentry Transaction. The backend request creates a new Sentry Transaction for its OpenTelemetry Span. The Sentry Transaction and Span are linked as a trace for navigation and error tracking.
```

Are you already familiar with OpenTelemetry but new to Sentry? Do you find our naming confusing? Here is a little introduction to help you apply your OpenTelemetry knowledge to Sentry.

> A `Span` generated by OpenTelemetry may be turned into a Sentry `Transaction` or `Span`. The first OpenTelemetry `Span` that is sent through the Sentry `SpanProcessor` is turned into a Sentry `Transaction`. Any children of this OpenTelemetry `Span `will be attached to this Sentry `Transaction` by looking at the parent span context of the OpenTelemetry `Span`. This happens for the OpenTelemetry root `Span` but also for any `Span` that is the top most `Span` on the current system. For example, you may send a request from your frontend to your backend. The frontend creates an OpenTelemetry root `Span` and a corresponding Sentry `Transaction`. For the outgoing request another OpenTelemetry `Span` is created as child of the root `Span`. Tracing information is passed alongside the request that is sent to the backend. The backend then creates a new Sentry `Transaction` for the OpenTelemetry `Span` that is created from the tracing information. A database call on the backend then creates another OpenTelemetry `Span` which is turned into a Sentry `Span` that is attached to the Sentry `Transaction`. The Sentry transactions created are linked together as a trace which allows you to navigate its transactions. In case an error happens it is linked to the `Transaction` showing you where in the trace the error occurred.
> 

Reviewer 2: ```suggestion
With Sentry’s OpenTelemetry SDK, an OpenTelemetry `Span` becomes a Sentry `Transaction` or `Span`. The first `Span` sent through the Sentry `SpanProcessor` is a `Transaction`, and any child `Span` gets attached to the first `Transaction` upon checking the parent `Span` context. This is true for the OpenTelemetry root `Span` and any top level `Span` in the system. For example, a request sent from frontend to backend will create an OpenTelemetry root `Span` with a corresponding Sentry `Transaction`. The backend request will create a new Sentry `Transaction` for the OpenTelemetry `Span`. The Sentry `Transaction` and `Span` are linked as a trace for navigation and error tracking purposes.

```

<PlatformSection notSupported={["java"]}>

## Install

<PlatformContent includePath="performance/opentelemetry-install" />

## Usage

<PlatformContent includePath="performance/opentelemetry-setup" />

</PlatformSection>

<PlatformSection supported={["java"]}>

## Using `sentry-opentelemetry-agent` With Auto Initialization

> If you use `sentry-opentelemetry-agent`, it will look for `SENTRY_DSN` and `SENTRY_PROPERTIES_FILE` environment variables to be defined, and then initialize Sentry automatically. Only your `DSN` and `tracesSampleRate` need to be configured.

Author: Thanks for the review. Kept the first sentence as is as it's not defining the env vars but rather checking whether they're defined.

### Debugging

<PlatformContent includePath="performance/opentelemetry-setup/with-java-agent/with-auto-init/debugging" />

### Turn Off Exporter Error Messages

<PlatformContent includePath="performance/opentelemetry-setup/with-java-agent/exporter-messages" />

## Using `sentry-opentelemetry-agent` Without Auto Initialization

You may also disable automatic initialization of Sentry in `sentry-opentelemetry-agent` by setting `SENTRY_AUTO_INIT=false` as environment variable. This means you will either have to use another Sentry integration that performs initialization, for example Spring Boot, or initialize Sentry manually.

### Install

<PlatformContent includePath="performance/opentelemetry-install/with-java-agent/without-auto-init" />

### Usage

<PlatformContent includePath="performance/opentelemetry-setup/with-java-agent/without-auto-init" />

### Debugging

<PlatformContent includePath="performance/opentelemetry-setup/with-java-agent/without-auto-init/debugging" />

### Turn Off Exporter Error Messages

<PlatformContent includePath="performance/opentelemetry-setup/with-java-agent/exporter-messages" />

## Using OpenTelemetry Without Any Java Agent

> If the Java Agent approach is not for you, you may also manually initialize OpenTelemetry. We have a separate dependency for this use case that allows you to reuse classes used by `sentry-opentelemetry-agent`.

Reviewer 2: ```suggestion
If the Java Agent approach isn't for you, you can manually initialize OpenTelemetry. We have a separate dependency for this use case that allows you to reuse classes used by `sentry-opentelemetry-agent`.
```

### Install

<PlatformContent includePath="performance/opentelemetry-install/without-java-agent" />

### Usage

> You will have to configure both OpenTelemetry and Sentry to see transactions in Sentry and have errors linked to transactions created by OpenTelemetry.
> 

Reviewer 2: ```suggestion
You'll have to configure both OpenTelemetry and Sentry to see transactions in Sentry and have errors linked to transactions created by OpenTelemetry.

```

<PlatformContent includePath="performance/opentelemetry-setup/without-java-agent" />

</PlatformSection>

## Filtering

> In case you want to filter out transactions before sending them to Sentry, for example to get rid of health checks, you can use <PlatformLink to="/configuration/options/#before-send-transaction"><PlatformIdentifier name="before-send-transaction" /></PlatformLink> to drop them. <PlatformLink to="/configuration/filtering/#using-platformidentifier-namebefore-send-transaction-">Here you can find an example on how to use it</PlatformLink>.

Reviewer 2: ```suggestion
If you want to filter out transactions before sending them to Sentry, ( to get rid of health checks, for example), you can use <PlatformLink to="/configuration/options/#before-send-transaction"><PlatformIdentifier name="before-send-transaction" /></PlatformLink> to drop them. <PlatformLink to="/configuration/filtering/#using-platformidentifier-namebefore-send-transaction-">Here's an example</PlatformLink>.
```
